File: appcommons/src/main/java/ng/kingsley/android/cache/Converter.java
package ng.kingsley.android.cache;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

/**
 * @author ADIO Kingsley O.
 * @since 14 Oct, 2016
 */

public interface Converter<T> {

    void writeToStream(OutputStream stream, T content) throws IOException;

    T readFromStream(InputStream stream) throws IOException;

}

File: preference/src/main/java/com/kingsleyadio/appcommons/preference/Converter.kt
package com.kingsleyadio.appcommons.preference

import java.lang.reflect.Type

/**
 * @author ADIO Kingsley O.
 * @since 14 Oct, 2016
 */
interface Converter {

    fun <T> toString(content: T, type: Type): String

    fun <T> fromString(serialized: String, type: Type): T?
}
